test/withdrawOptionRequest.ts

```diff
@@ -18,8 +18,23 @@ describe("NFTOpt Tests", function () {
     });
 
     describe("withdrawOptionRequest", function () {
-        it("tests that method can be called", async function () {
-            expect(NFTOptCTR.connect(buyer).withdrawOptionRequest(0)).to.not.throw;
+
+        it("should fail when option does not exist", async function () {
+            await expect(NFTOptCTR.connect(buyer)
+                .withdrawOptionRequest(9999))
+                .to.be.revertedWith("INVALID_OPTION_ID");
+        });
+
+        it("should fail when option is not in REQUEST state", async function () {
+            await publishDummyOptionRequest();
+            const option = await NFTOptCTR.options(1);
+            expect(dummyOptionRequest.state).to.equal(OptionState.Request);
```

Comment:
if the test describes it as `should fail when...` then the check should be in the same manner, to test what you are describing => it should make use of ....to.be.revertedWith('your_custom_error`), just as you did above.

same for the test below

Reply:
but still, the check is faulty in the same way as before, because it uses the `dummyOptionRequest`, not the one that you actually published above